### PR TITLE
feat(pubsub): add generated admin api

### PR DESCRIPTION
**Pull request by alevenberg**
Part of https://github.com/googleapis/google-cloud-cpp/issues/12873

A note about the generated changes. It uses the service comment which says the client does both data and admin ops. Do we want to omit this and substitute our own?

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12933)

### CODE REVIEW

File: generator/generator_config.textproto [hunk 1]
```diff
@@ -2550,6 +2550,26 @@ service {
   omit_repo_metadata: true
 }
 
+# Cloud Pub/Sub Generated Admin API
+service {
+  service_proto_path: "google/pubsub/v1/pubsub.proto"
+  product_path: "google/cloud/pubsub/admin"
+  initial_copyright_year: "2023"
+  retryable_status_codes: ["kUnavailable"]
+  generate_round_robin_decorator: true
```
Comment (coryan) [inline generator/generator_config.textproto:2559]: I do not think we need this for the admin APIs.

File: google/cloud/pubsub/admin/subscription_admin_client.h [hunk 2]
```diff
@@ -0,0 +1,925 @@
...
+
+namespace google {
+namespace cloud {
+namespace pubsub_admin {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+///
+/// The service that an application uses to manipulate subscriptions and to
+/// consume messages from a subscription via the `Pull` method or by
+/// establishing a bi-directional stream using the `StreamingPull` method.
```
Comment (coryan) [inline google/cloud/pubsub/admin/subscription_admin_client.h:39]: I see what you mean.  Yes, we need to figure out a way to replace the documentation for this class.

File: google/cloud/pubsub/admin/subscription_admin_client.h [hunk 3]
```diff
@@ -0,0 +1,925 @@
...
+  ///  rules](https://cloud.google.com/pubsub/docs/admin#resource_names). Format
+  ///  is `projects/{project}/snapshots/{snap}`.
+  /// @param subscription  Required. The subscription whose backlog the snapshot retains.
+  ///  Specifically, the created snapshot is guaranteed to retain:
+  ///   (a) The existing backlog on the subscription. More precisely, this is
+  ///       defined as the messages in the subscription's backlog that are
+  ///       unacknowledged upon the successful completion of the
+  ///       `CreateSnapshot` request; as well as:
+  ///   (b) Any messages published to the subscription's topic following the
+  ///       successful completion of the CreateSnapshot request.
```
Comment (coryan) [inline google/cloud/pubsub/admin/subscription_admin_client.h:681]: FYI, some of this documentation may not render as well as we would like.  Nothing we can do, maybe just look at it and then ask or send fixes upstream.

**Review by coryan — COMMENTED** [review]
Yes, it does look like we need to fix the comments.  One quick hack would be to add them to the other hacks here:

https://github.com/googleapis/google-cloud-cpp/blob/3c1a946c9f1c6a5be232064af8cfa9747a7f807e/generator/internal/descriptor_utils.cc#L266-L280

But it would be nicer to have an option with a "replacement class comment" or something.

**Comment by alevenberg** [comment]
Closing since I will need to regenerate based on new generator changes.

File: google/cloud/pubsub/admin/subscription_admin_client.h (hunk 2 above)
Comment (alevenberg) [reply google/cloud/pubsub/admin/subscription_admin_client.h:39]: https://github.com/googleapis/google-cloud-cpp/pull/12982

File: google/cloud/pubsub/admin/subscription_admin_client.h (hunk 3 above)
Comment (coryan) [reply google/cloud/pubsub/admin/subscription_admin_client.h:681]: I would like it to be rendered as the service team intended.  I this case I guess they wanted an enumerated list, but I am not sure markdown will represent it as such.  I think most markdown engines will treat that text as a single sentence.

Proto comments are supposed to be CommonMark:

https://google.aip.dev/192#formatting

But not all service teams follow that guidance, and the results are hilarious sometimes.  This one in particular does not look like CommonMark, but I might be wrong.

PS: even if it is or becomes formatted in CommonMark "standard": the C++ tooling and/or the tooling for other languages may mess things up as it goes through 1 or 2 markdown engines that may not perfectly follow the  CommonMark guidelines in all cases.

Comment (alevenberg) [reply google/cloud/pubsub/admin/subscription_admin_client.h:681]: Hm.. my interpretation was this wasn't markdown at all. But they are using [backticks](https://github.com/googleapis/googleapis/blob/master/google/pubsub/v1/pubsub.proto#L1440-L1448)...

We can see how it looks and try and fix later maybe, but sounds like change is not blocking. SGTM!

**Pull request closed without merging**